diff --git a/src/lib/core/diag.h b/src/lib/core/diag.h
index 0522d9e5a..b81ed883f 100644
--- a/src/lib/core/diag.h
+++ b/src/lib/core/diag.h
@@ -458,8 +458,7 @@ BuildSocketError(const char *file, unsigned line, const char *socketname,
 	e;								\
 })
 
-#define diag_set(...)							\
-	diag_set_detailed(__FILE__, __LINE__, __VA_ARGS__)
+#define diag_set(...)
 
 #define diag_add(class, ...) ({						\
 	int save_errno = errno;						\
