diff --git a/caffe2/serialize/inline_container.cc b/caffe2/serialize/inline_container.cc
index 9f0e9ce619..c70f51161e 100644
--- a/caffe2/serialize/inline_container.cc
+++ b/caffe2/serialize/inline_container.cc
@@ -128,7 +128,15 @@ void PyTorchStreamReader::init() {
     std::tie(version_ptr, version_size) = getRecord("version");
   }
   std::string version(static_cast<const char*>(version_ptr.get()), version_size);
-  version_ = caffe2::stoull(version);
+  // version_ = caffe2::stoull(version);
+  try {
+    version_ = caffe2::stoull(version);
+  } catch(const std::invalid_argument &e) {
+    CAFFE_THROW(
+        "stoull exception intercepted for invalid version = <",
+        version,
+        ">");
+  }
   AT_ASSERTM(
       // NOLINTNEXTLINE(clang-diagnostic-sign-compare)
       version_ >= kMinSupportedFileFormatVersion,
