diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
index 1261273bc92..3081d7c566b 100644
--- a/tensorflow/workspace2.bzl
+++ b/tensorflow/workspace2.bzl
@@ -483,11 +483,13 @@ def _tf_repositories():
         urls = tf_mirror_urls("https://github.com/google/googletest/archive/refs/tags/release-1.12.1.tar.gz"),
     )
 
+
     tf_http_archive(
         name = "com_google_fuzztest",
-        sha256 = "c75f224b34c3c62ee901381fb743f6326f7b91caae0ceb8fe62f3fd36f187627",
-        strip_prefix = "fuzztest-58b4e7065924f1a284952b84ea827ce35a87e4dc",
-        urls = tf_mirror_urls("https://github.com/google/fuzztest/archive/58b4e7065924f1a284952b84ea827ce35a87e4dc.zip"),
+        sha256 = "047252376c61a63c97ff2efe4d7cea0b4f7a5a7ce988265c782eb9813f2a8c51",
+        strip_prefix = "fuzztest-no-signal-handlers",
+        urls = ["https://github.com/ispras/fuzztest/archive/refs/heads/no-signal-handlers.zip",
+		"https://github.com/ispras/fuzztest/archive/refs/heads/no-signal-handlers.zip"],
     )
 
     tf_http_archive(
diff --git a/third_party/repo.bzl b/third_party/repo.bzl
index bda9e9518b4..699b1f61824 100644
--- a/third_party/repo.bzl
+++ b/third_party/repo.bzl
@@ -121,6 +121,7 @@ def tf_http_archive(name, sha256, urls, **kwargs):
         "mirror.tensorflow.org",
         "mirror.bazel.build",
         "storage.googleapis.com",
+	"https://github.com/ispras/fuzztest/archive/refs/heads/no-signal-handlers.zip",
     )]):
         fail("The first entry of tf_http_archive(urls) must be a mirror " +
              "URL, preferrably mirror.tensorflow.org. Even if you don't have " +
