diff --git a/tensorflow/security/fuzzing/cc/BUILD b/tensorflow/security/fuzzing/cc/BUILD
index bf419d8332c..fc453ae53a7 100644
--- a/tensorflow/security/fuzzing/cc/BUILD
+++ b/tensorflow/security/fuzzing/cc/BUILD
@@ -58,24 +58,24 @@ tf_cc_fuzz_test(
     ],
 )
 
-tf_cc_fuzz_test(
-    name = "checkpoint_reader_fuzz",
-    srcs = ["checkpoint_reader_fuzz.cc"],
-    data = glob(["checkpoint_reader_testdata/*"]),
-    tags = ["no_oss"],
-    deps = [
-        ":checkpoint_reader_fuzz_input_proto_cc",
-        "//tensorflow/c:checkpoint_reader",
-        "//tensorflow/c:tf_status",
-        "//tensorflow/c:tf_status_helper",
-        "//tensorflow/core:framework",
-        "//tensorflow/core:lib",
-        "//tensorflow/core/framework:types_proto_cc",
-        "//tensorflow/core/platform:resource_loader",
-        "//tensorflow/core/util:saved_tensor_slice_proto_cc",
-        "@local_xla//xla/tsl/platform:status",
-    ],
-)
+#tf_cc_fuzz_test(
+#    name = "checkpoint_reader_fuzz",
+#    srcs = ["checkpoint_reader_fuzz.cc"],
+#    data = glob(["checkpoint_reader_testdata/*"]),
+#    tags = ["no_oss"],
+#    deps = [
+#        ":checkpoint_reader_fuzz_input_proto_cc",
+#        "//tensorflow/c:checkpoint_reader",
+#        "//tensorflow/c:tf_status",
+#        "//tensorflow/c:tf_status_helper",
+#        "//tensorflow/core:framework",
+#        "//tensorflow/core:lib",
+#        "//tensorflow/core/framework:types_proto_cc",
+#        "//tensorflow/core/platform:resource_loader",
+#        "//tensorflow/core/util:saved_tensor_slice_proto_cc",
+#        "@local_xla//xla/tsl/platform:status",
+#    ],
+#)
 
 tf_proto_library(
     name = "checkpoint_reader_fuzz_input_proto",
@@ -218,27 +218,27 @@ cc_library(
     ],
 )
 
-tf_cc_fuzz_test(
-    name = "end_to_end_fuzz",
-    srcs = ["end_to_end_fuzz.cc"],
-    deps = [
-        "//tensorflow/cc/saved_model:constants",
-        "//tensorflow/cc/saved_model:loader",
-        "//tensorflow/cc/saved_model:tag_constants",
-        "//tensorflow/core:core_cpu",
-        "//tensorflow/core:lib",
-        "//tensorflow/core:protos_all_cc",
-        "//tensorflow/core:tensorflow",
-        "//tensorflow/core/framework:tensor",
-        "//tensorflow/core/platform:status",
-        "//tensorflow/security/fuzzing/cc/core/framework:datatype_domains",
-        "//tensorflow/security/fuzzing/cc/core/framework:tensor_domains",
-        "//tensorflow/security/fuzzing/cc/core/framework:tensor_shape_domains",
-        "@com_google_absl//absl/status",
-        "@local_xla//xla/tsl/platform:env",
-        "@local_xla//xla/tsl/platform:status",
-    ],
-)
+#tf_cc_fuzz_test(
+#    name = "end_to_end_fuzz",
+#    srcs = ["end_to_end_fuzz.cc"],
+#    deps = [
+#        "//tensorflow/cc/saved_model:constants",
+#        "//tensorflow/cc/saved_model:loader",
+#        "//tensorflow/cc/saved_model:tag_constants",
+#        "//tensorflow/core:core_cpu",
+#        "//tensorflow/core:lib",
+#        "//tensorflow/core:protos_all_cc",
+#        "//tensorflow/core:tensorflow",
+#        "//tensorflow/core/framework:tensor",
+#        "//tensorflow/core/platform:status",
+#        "//tensorflow/security/fuzzing/cc/core/framework:datatype_domains",
+#        "//tensorflow/security/fuzzing/cc/core/framework:tensor_domains",
+#        "//tensorflow/security/fuzzing/cc/core/framework:tensor_shape_domains",
+#        "@com_google_absl//absl/status",
+#        "@local_xla//xla/tsl/platform:env",
+#        "@local_xla//xla/tsl/platform:status",
+#    ],
+#)
 
 tf_cc_fuzz_test(
     name = "text_literal_reader_fuzz",
diff --git a/tensorflow/security/fuzzing/tf_fuzzing.bzl b/tensorflow/security/fuzzing/tf_fuzzing.bzl
index d8a6bafbf10..7e7582d9caa 100644
--- a/tensorflow/security/fuzzing/tf_fuzzing.bzl
+++ b/tensorflow/security/fuzzing/tf_fuzzing.bzl
@@ -4,6 +4,26 @@ load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("@rules_python//python:py_test.bzl", "py_test")
 
 # TensorFlow fuzzing can be done in open source too, as it is in oss-fuzz.com
+load(
+    "//tensorflow:tensorflow.bzl",
+    "tf_cc_test",
+)
+
+def tf_oss_fuzz_fuzztest(
+        name,
+        fuzzing_dict = [],
+        corpus = [],
+        deps = [],
+        tags = [],
+        **kwargs):
+    tf_cc_test(
+        name = name,
+        deps = deps + [
+            "@com_google_fuzztest//fuzztest",
+            "@com_google_fuzztest//fuzztest:fuzztest_gtest_main",
+        ],
+        **kwargs
+    )
 
 # tf_cc_fuzz_test is a cc_test modified to include fuzzing support and dependencies for go/fuzztest.
 def tf_cc_fuzz_test(
