--- node.gyp	2023-01-13 17:19:02.673618865 +0300
+++ node_libfuzzer.gyp	2023-01-13 17:14:57.874849827 +0300
@@ -891,7 +891,7 @@
       ],
     }, # node_lib_target_name
     { # fuzz_url
-      'target_name': 'fuzz_url',
+      'target_name': 'load_url_fuzzer',
       'type': 'executable',
       'dependencies': [
         '<(node_lib_target_name)',
@@ -913,7 +913,7 @@
       ],
       'conditions': [
         ['OS=="linux"', {
-          'ldflags': [ '-fsanitize=fuzzer' ]
+          'ldflags': ['-g -std=c++17 -fsanitize=fuzzer,address,integer,undefined,bounds,null,float-divide-by-zero' ]
         }],
         # Ensure that ossfuzz flag has been set and that we are on Linux
         [ 'OS!="linux" or ossfuzz!="true"', {
@@ -922,7 +922,7 @@
       ],
     }, # fuzz_url
     { # fuzz_env
-      'target_name': 'fuzz_env',
+      'target_name': 'load_env_fuzzer',
       'type': 'executable',
       'dependencies': [
         '<(node_lib_target_name)',
@@ -952,7 +952,7 @@
       ],
       'conditions': [
         ['OS=="linux"', {
-          'ldflags': [ '-fsanitize=fuzzer' ]
+          'ldflags': ['-g -std=c++17 -fsanitize=fuzzer,address,integer,undefined,bounds,null,float-divide-by-zero' ]
         }],
         # Ensure that ossfuzz flag has been set and that we are on Linux
         [ 'OS!="linux" or ossfuzz!="true"', {
