diff --git a/cipher.c b/cipher.c
index f12daa1..d36b44e 100644
--- a/cipher.c
+++ b/cipher.c
@@ -75,7 +75,7 @@ struct sshcipher {
 #define CFLAG_CHACHAPOLY	(1<<1)
 #define CFLAG_AESCTR		(1<<2)
 #define CFLAG_NONE		(1<<3)
-#define CFLAG_INTERNAL		CFLAG_NONE /* Don't use "none" for packets */
+#define CFLAG_INTERNAL 0
 #ifdef WITH_OPENSSL
 	const EVP_CIPHER	*(*evptype)(void);
 #else
diff --git a/includes.h b/includes.h
index 6d17ef6..ff5d31e 100644
--- a/includes.h
+++ b/includes.h
@@ -111,7 +111,7 @@
 #include <netinet/in.h>
 #include <netinet/in_systm.h> /* For typedefs */
 #ifdef HAVE_RPC_TYPES_H
-# include <rpc/types.h> /* For INADDR_LOOPBACK */
+//# include <rpc/types.h> /* For INADDR_LOOPBACK */
 #endif
 #ifdef USE_PAM
 #if defined(HAVE_SECURITY_PAM_APPL_H)
diff --git a/ssh-agent.c b/ssh-agent.c
index d35741a..657ee12 100644
--- a/ssh-agent.c
+++ b/ssh-agent.c
@@ -1495,7 +1495,7 @@ process_lock_agent(SocketEntry *e, int lock)
 			delay = 100000 * fail_count;
 			debug("unlock failed, delaying %0.1lf seconds",
 			    (double)delay/1000000);
-			usleep(delay);
+			// usleep(delay);
 		}
 		explicit_bzero(passwdhash, sizeof(passwdhash));
 	} else if (!locked && lock) {
