diff --git a/cmake/TorchAudioHelper.cmake b/cmake/TorchAudioHelper.cmake
index d000483e..3a22ca1c 100644
--- a/cmake/TorchAudioHelper.cmake
+++ b/cmake/TorchAudioHelper.cmake
@@ -1,21 +1,7 @@
 find_package(Torch REQUIRED)
 
-# Remove stray mkl dependency found in Intel mac.
-#
-# For Intel mac, torch_cpu has caffe2::mkl, which adds link flags like
-# -lmkl_intel_ilp64, -lmkl_core and -lmkl_intel_thread.
-# Even though TorchAudio does not call any of MKL functions directly,
-# Apple's linker does not drop them, instead it bakes these dependencies
-# Therefore, we remove it.
-# See https://github.com/pytorch/audio/pull/3307
-get_target_property(dep torch_cpu INTERFACE_LINK_LIBRARIES)
-if ("caffe2::mkl" IN_LIST dep)
-  list(REMOVE_ITEM dep "caffe2::mkl")
-  set_target_properties(torch_cpu PROPERTIES INTERFACE_LINK_LIBRARIES "${dep}")
-endif()
-
 function (_library destination name source include_dirs link_libraries compile_defs)
-  add_library(${name} SHARED ${source})
+  add_library(${name} STATIC ${source})
   target_include_directories(${name} PRIVATE "${PROJECT_SOURCE_DIR}/src;${include_dirs}")
   target_link_libraries(${name} ${link_libraries})
   target_compile_definitions(${name} PRIVATE ${compile_defs})
@@ -62,7 +48,7 @@ if (BUILD_TORCHAUDIO_PYTHON_EXTENSION)
     set(ADDITIONAL_ITEMS Python3::Python)
   endif()
   function(_extension destination name sources include_dirs libraries definitions)
-    add_library(${name} SHARED ${sources})
+    add_library(${name} STATIC ${sources})
     target_compile_definitions(${name} PRIVATE "${definitions}")
     target_include_directories(
       ${name}
diff --git a/src/libtorchaudio/sox/CMakeLists.txt b/src/libtorchaudio/sox/CMakeLists.txt
index 5ffe782c..4c03bd4f 100644
--- a/src/libtorchaudio/sox/CMakeLists.txt
+++ b/src/libtorchaudio/sox/CMakeLists.txt
@@ -10,7 +10,7 @@ torchaudio_library(
   libtorchaudio_sox
   "${sources}"
   ""
-  "torch;sox"
+  "torch;/usr/local/lib/libsox.a"
   ""
   )
 
diff --git a/src/libtorchaudio/sox/io.h b/src/libtorchaudio/sox/io.h
index b011ef59..ce2da7c0 100644
--- a/src/libtorchaudio/sox/io.h
+++ b/src/libtorchaudio/sox/io.h
@@ -1,7 +1,7 @@
 #ifndef TORCHAUDIO_SOX_IO_H
 #define TORCHAUDIO_SOX_IO_H
 
-#include <libtorchaudio/sox/utils.h>
+#include "utils.h"
 #include <torch/script.h>
 
 namespace torchaudio::sox {
